packages/block-library/src/query/utils.js

```diff
@@ -214,3 +215,43 @@ export const getTransformedBlocksFromPattern = (
 	}
 	return { newBlocks: clonedBlocks, queryClientIds };
 };
+
+/**
+ * Helper hook that determines if there is an active variation of the block
+ * and if there are available specific patterns for this variation.
+ * If there are, they are going to be suggested to the user in setup and
+ * replace flow.
+ *
+ * If there are no patterns, the default ones for Query Loop are going to be suggested.
+ *
+ * @param {string} clientId   The block's client ID.
+ * @param {Object} attributes The block's attributes.
+ * @return {string} The block name to be used in the patterns suggestions.
+ */
+export function useBlockNameForPatterns( clientId, attributes ) {
+	const activeVariationName = useSelect(
```

Comment:
Can we end up with a undefined `activeVariationName` if none is found? I'm not super familiar with all these flows to tell how to reproduce it but wanted to bring it up in case anyone can clarify it.

Reply:
Ah, I see. We can by unregistering all block variations. In that case we'll be looking for patterns targeted at `core/query/undefined` and none will be found, hence we'll return the `core/query` name anyway. It works, though a micro-optimization would be to return early if `activeVariationName` is `undefined`.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/44197/commits/6e0c5d2ade2baa3afab2d3919e84c96982f948ec

Reply:
At the risk of picking nits: wouldn't it be enough to return early instead of with a cached array?

```diff
-return EMPTY_ARRAY;
+return;
```

The final return statement in the hook already comes with optional chaining: _activeVariationPatterns **?** .length_

Reply:
haha, you're right! I updated and it's better now. Thanks both!